firedrake/mesh.py

```diff
@@ -106,6 +106,15 @@ def subset(self, markers):
         try:
             return self._subsets[markers]
         except KeyError:
+            # check that the given markers are valid
+            for marker in markers:
+                if marker not in self.markers:
```

Comment:
I have tested this. In

    tests/extrusion/test_steady_advection_3D_extr.py

the function *test_near_to_far* fails: "3 is not a valid marker"

I also tested the following. I included

    if markers is not None:
        for marker in markers:
             assert(marker in unique_markers)

 into *__init__* of the *_Facets* class. Which also fails. Thus *unique_markers* seems not always to be a superset of *markers*.